File: app/src/main/java/com/hfad/mymovies/data/models/ExampleResponse.java
package com.hfad.mymovies.data.models;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class ExampleResponse {
    @SerializedName("page")
    @Expose
    private int page;
    @SerializedName("total_results")
    @Expose
    private int totalResults;
    @SerializedName("total_pages")
    @Expose
    private int totalPages;
    @SerializedName("results")
    @Expose
    private List<MovieDiscover> results = null;
    @SerializedName("id")
    @Expose
    private int id;
    @SerializedName("cast")
    @Expose
    private List<Actors> cast = null;
    @SerializedName("crew")
    @Expose
    private List<Crew> crew = null;
    @SerializedName("genres")
    @Expose
    private List<Genre> genres = null;

    public List<Genre> getGenres() {
        return genres;
    }

    public void setGenres(List<Genre> genres) {
        this.genres = genres;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public List<Actors> getCast() {
        return cast;
    }

    public void setCast(List<Actors> cast) {
        this.cast = cast;
    }

    public List<Crew> getCrew() {
        return crew;
    }

    public void setCrew(List<Crew> crew) {
        this.crew = crew;
    }

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;
    }

    public int getTotalResults() {
        return totalResults;
    }

    public void setTotalResults(int totalResults) {
        this.totalResults = totalResults;
    }

    public int getTotalPages() {
        return totalPages;
    }

    public void setTotalPages(int totalPages) {
        this.totalPages = totalPages;
    }

    public List<MovieDiscover> getResults() {
        return results;
    }

    public void setResults(List<MovieDiscover> results) {
        this.results = results;
    }
}

File: app/src/main/java/com/hfad/mymovies/data/models/ExampleResponse.kt
package com.hfad.mymovies.data.models

import com.google.gson.annotations.SerializedName

data class ExampleResponse (
    @SerializedName("page")
    var page: Int = 0,
    @SerializedName("total_results")
    var totalResults: Int = 0,
    @SerializedName("total_pages")
    var totalPages: Int = 0,
    @SerializedName("results")
    var results: List<MovieDiscover>? = null,
    @SerializedName("id")
    var id: Int = 0,
    @SerializedName("cast")
    var cast: List<Actors>? = null,
    @SerializedName("crew")
    var crew: List<Crew>? = null,
    @SerializedName("genres")
    var genres: List<Genre>? = null
)